junction/src/main/java/org/killbill/billing/junction/plumbing/billing/DefaultInternalBillingApi.java

```diff
@@ -295,7 +295,7 @@ private void addBillingEventsForSubscription(final ImmutableAccountData account,
             BillingAlignment alignment = null;
             for (final SubscriptionBillingEvent transition : billingTransitions) {
 
-                if (transition.getType() != SubscriptionBaseTransitionType.CANCEL) {
+                if (transition.getType() != SubscriptionBaseTransitionType.CANCEL && transition.getType() != SubscriptionBaseTransitionType.EXPIRED) { //TODO_1533 Added to prevent NPE while running TestCatalogFixedTerm tests after notification processing changes, revisit
```

Comment:
I think this is correct, you can remove the `TODO_1533`. The idea is that if we reach the end of a subscription, whether through a cancelation or an expiration, we don't need to compute the effective BCD (BillCycleDay) for this transition as there is nothing to bill

Reply:
Fixed